Add msgraph to SLFS.

renodr opened the pull request:
This package is useful for students and remote workers who need to interact with the Microsoft ecosystem. OneDrive support has been added as default in Windows 11 for a long time now, so this allows users on Linux to be able to access files from Windows machines that get automatically backed up to OneDrive.

Users who install this will need to rebuild gvfs in BLFS without the -Donedrive=false option. I've added some basic usage information into the page as well to that effect!

Fixes #177

renodr committed:
Add msgraph to SLFS.

This package is useful for students and remote workers who need to
interact with the Microsoft ecosystem. OneDrive support has been added
as default in Windows 11 for a long time now, so this allows users on
Linux to be able to access files from Windows machines that get
automatically backed up to OneDrive.

Users who install this will need to rebuild gvfs in BLFS without the
-Donedrive=false option. I've added some basic usage information into
the page as well to that effect!

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,138 @@
+  <!ENTITY % general-entities SYSTEM "../../general.ent">
+  %general-entities;
+
+  <!ENTITY gnome-download-http  "https://download.gnome.org/sources">
```
We have `/gnome.ent`. The problem is that it isn't included anywhere. It'd be a better idea to remove this line and include that file in `/general.ent` so we don't have to define it the download root once more.

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,138 @@
+<screen><userinput>mkdir build &amp;&amp;
+cd    build &amp;&amp;
+
+meson setup --prefix=/usr --buildtype=release -Dtests=false .. &amp;&amp;
```
Separate `-D` from `tests=false` like `-D tests=false`. Make sure all instances of `-Dtests=false` that proceed get changed to `-D tests=false`.

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,138 @@
+      program
+      <ulink url="https://gitlab.freedesktop.org/pwithnall/uhttpmock">uhttpmock</ulink>.
+      If you want to run the tests, install uhttpmock and remove -Dtests=false,
+      and then run <command>ninja test</command>
```
Put a `.` after `</command>`, like so: `<command>ninja test</command>.`

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,138 @@
+    <para>
+      To configure msgraph to allow access to Microsoft OneDrive with packages
+      that use gvfs, you should rebuild gvfs in BLFS without the
+      <parameter>-Donedrive=false</parameter> option. After doing this, you
```
`-Donedrive=false` -> `-D onedrive=false`.

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,138 @@
+            Microsoft Graph API.
+          </para>
+          <indexterm zone="msgraph libmsgraph">
+            <primary sortas="c-libmsgraph">libmsgraph-1.so</primary>
```
In GLFS and SLFS, we deviate from BLFS practices and align more with LFS. Library names listed skip the suffix. If the library is a different format than an LSB Shared Object (.SO), specify as such (static, Java library, DLL, etc.) So remove all instances of `.so` to align with that practice.

zeckma-tech commented on introduction/welcome/changelog.xml:
```diff
@@ -43,6 +43,9 @@
       <para>August 6th, 2025</para>
       <itemizedlist>
+        <listitem>
+          <para>[renodr] - msgraph: Added.</para>
```
Include the issue number, saying this change fixes it. PR numbers aren't usually included anymore in changelog entries but you can also add it if you wish. But the issue number should be there.

Zeckmathederg reviewed (changes requested):
Some things here and there. All are minor and aren't showstoppers but it'd be nice seeing these changed!

renodr committed:
msgraph: Minor tweaks requested by Zeckma.

- Moved gnome-download-http to general.ent
- Added spaces between meson options
- Added a period at the end of a line
- Dropped .so from library descriptions
- Included the issue number in the changelog

renodr commented:
Thank you for the review! I've made the changes as requested :smiley:

zeckma-tech commented on general.ent:
```diff
@@ -93,6 +93,8 @@
 <!ENTITY w3m-url              "http://w3m.sourceforge.net/">
 <!ENTITY sysprof-url          "https://wiki.gnome.org/Apps/Sysprof">
 <!ENTITY man                  "https://man.archlinux.org/man/">
+<!ENTITY gnome-download-http  "https://download.gnome.org/sources">
```
No, we should include `gnome.ent` in this file, not the download root itself, since then we'd be redefining the variable that's already defined in a file that isn't included but should be. We're at a point now where we actually have a use for `gnome.ent`, so we should now include it.

zeckma-tech commented on general/genlib/msgraph.xml:
```diff
@@ -0,0 +1,137 @@
+      This package does come with a test suite, but it requires the external
+      program
+      <ulink url="https://gitlab.freedesktop.org/pwithnall/uhttpmock">uhttpmock</ulink>.
+      If you want to run the tests, install uhttpmock and remove -D tests=false,
```
Didn't notice this, but we should also surround `-D tests=false` with `<parameter>` tags, so: `<parameter>-D tests=false</parameter>`. Of course, make sure line count doesn't exceed 80 characters.

Zeckmathederg reviewed (changes requested):
Two minor changes again. One I overlooked (my bad!) but also the Gnome download link root needs changing (see the change request below).

renodr committed:
msgraph: A couple more minor tweaks :)

- Include gnome.ent in general.ent instead of redefining
  gnome-download-http
- Surround "-D tests=false" in the test suite information with
  <parameter>

renodr committed:
Fix an XML syntax error that crept in after I built the book

renodr commented:
Thank you again for the review!

Zeckmathederg reviewed (approved):
Okay, this looks good to me now. Thank you for this PR and for working with me to resolve those minor issues beforehand! Merging.
